buildSrc/src/main/kotlin/java-library-conventions.gradle.kts

```diff
@@ -180,6 +183,29 @@ tasks.jar {
 				"Implementation-Vendor" to "junit.org"
 		)
 	}
+
+	withConvention(BundleTaskConvention::class) {
+		setBnd("""
+			Bundle-Name: ${project.description}
+			Import-Package: \
+				!org.apiguardian.api,\
+				kotlin.*;resolution:="optional",\
+				*
+			-fixupmessages.kotlin.import: "Unused Import-Package instructions: \\[kotlin.*\\]";is:=ignore
+			-removeheaders: Private-Package
+			-noextraheaders: true
+
+			# The following exports only packages having classes annotated @API
+			# First collect all classes annotated with @API
+			possibleAPI: ${'$'}{classes;HIERARCHY_INDIRECTLY_ANNOTATED;org.apiguardian.api.API}
+			# Strip off the classNames to obtain a list of package names
+			possiblePackages: ${'$'}{replace;${'$'}{possibleAPI};\.[^.]*${'$'};}
```

Comment:
I'm curious if we could we parse the module descriptors instead... 🤔

Reply:
That would be great it we could!

Better to not duplicate this information.

Reply:
We could parse their source code (similar to [this](https://github.com/sormuras/bach/blob/master/src/bach/Bach.java#L697-L728)) or use the already compiled descriptors -- they are compiled at this stage, aren't they?

Reply:
The module descriptor will always trail your actual code. You should let tooling give you a result _as close to perfect as a algorithm can bring it_ and then let you take it the last mile. This will eliminate guess work and automatically find details humans aren't good at identifying.

In short, let's leave that part for a future update, shall we? :)

Reply:
This is not to say that I will not solve the @API exports issue... but rather then parsing of the module descriptor suggestion. ;)

Reply:
... if that's ok with you all!

Reply:
Fair enough, we could instead add an integration test that verifies the two things are aligned.

Reply:
resolving this since we have a solution now